### DISCUSSION TITLE
Attribute rules not working

### DISCUSSION THREAD

**Discussion by michaelcsikos** [MarimerLLC/csla#1784] [Questions] [answered]
I have two solutions which are set up virtually identically. The EXE and WinForms DLLs are .NET 4.8, the other layers are .NET Standard 2.0. It's (still) using CSLA 4.11.2. In one solution attribute rules, e.g. `[Required]`, are not working. I have copied the same class to both solutions. It works in one and not the other. The `app.config` and `bindingRedirects` are identical. Any ideas?

**Comment by rockfordlhotka** [comment, marked as answer]
Have you confirmed that the dependencies (like `System.ComponentModel.Annotations`) are identical - same versions, same NuGet package names?

**Reply by michaelcsikos** [reply]
Thanks for your quick reply. In the output folder `System.ComponentModel.Annotations` are indeed different. The projects weren't explicitly bringing them in, but the project where the rules work is using `Microsoft.EntityFrameworkCore` which depends on version 4.7.0. Adding 4.7.0 to the Business layer in the faulty solution via NuGet solves it.
